Publishing with a rid can cause dependency package downgrade errors (NU1605)

For example, this generated project:
```xml
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <TargetFramework>netcoreapp3.1</TargetFramework>
        <LangVersion>preview</LangVersion>
        <Configuration>Release</Configuration>
        <AssemblyName>Microsoft.Azure.Functions.Worker.Extensions</AssemblyName>
        <RootNamespace>Microsoft.Azure.Functions.Worker.Extensions</RootNamespace>
        <MajorMinorProductVersion>1.0</MajorMinorProductVersion>
        <Version>$(MajorMinorProductVersion).0</Version>
        <AssemblyVersion>$(MajorMinorProductVersion).0.0</AssemblyVersion>
        <FileVersion>$(Version)</FileVersion>
        <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>        
    </PropertyGroup>
    <ItemGroup>
        <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="3.0.11" />
        <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Storage" Version="4.0.4" />

        <!-- If you add a ref to the package below... the error goes away... -->
        <!--<PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Kafka" Version="3.2.1" />-->
    </ItemGroup>
</Project>
```

built with 
```
dotnet publish -r win-x86
```

Results in the errors:
```
 error NU1605: Detected package downgrade: System.Net.NameResolution from 4.3.0 to 4.0.0. Reference the package directly from the project to select a different version.
C:\Users\brettsam\AppData\Local\Temp\iv1phtlm.nwz\WorkerExtensions.csproj : error NU1605:  Microsoft.Azure.Functions.Worker.Extensions -> Microsoft.Azure.WebJobs.Extensions.Storage 4.0.4 -> Microsoft.Azure.Cosmos.Table 1.0.7 -> Microsoft.Azure.DocumentDB.Core 2.10.0 -> System.Net.NetworkInformation 4.1.0 -> System.Net.Sockets 4.1.0 -> runtime.win.System.Net.Sockets 4.3.0 -> System.Net.NameResolution (>= 4.3.0)
C:\Users\brettsam\AppData\Local\Temp\iv1phtlm.nwz\WorkerExtensions.csproj : error NU1605:  Microsoft.Azure.Functions.Worker.Extensions -> Microsoft.Azure.WebJobs.Extensions.Storage 4.0.4 -> Microsoft.Azure.Cosmos.Table 1.0.7 -> Microsoft.Azure.DocumentDB.Core 2.10.0 -> System.Net.NameResolution (>= 4.0.0)
C:\Users\brettsam\AppData\Local\Temp\iv1phtlm.nwz\WorkerExtensions.csproj : error NU1605: Detected package downgrade: System.Security.Principal.Windows from 4.3.0 to 4.0.0. Reference the package directly from the project to select a different version.
C:\Users\brettsam\AppData\Local\Temp\iv1phtlm.nwz\WorkerExtensions.csproj : error NU1605:  Microsoft.Azure.Functions.Worker.Extensions -> Microsoft.Azure.WebJobs.Extensions.Storage 4.0.4 -> Microsoft.Azure.Cosmos.Table 1.0.7 -> Microsoft.Azure.DocumentDB.Core 2.10.0 -> System.Net.NetworkInformation 4.1.0 -> System.Net.Sockets 4.1.0 -> runtime.win.System.Net.Sockets 4.3.0 -> System.Security.Principal.Windows (>= 4.3.0)
C:\Users\brettsam\AppData\Local\Temp\iv1phtlm.nwz\WorkerExtensions.csproj : error NU1605:  Microsoft.Azure.Functions.Worker.Extensions -> Microsoft.Azure.WebJobs.Extensions.Storage 4.0.4 -> Microsoft.Azure.Cosmos.Table 1.0.7 -> Microsoft.Azure.DocumentDB.Core 2.10.0 -> System.Net.NetworkInformation 4.1.0 -> System.Security.Principal.Windows (>= 4.0.0)
C:\Users\brettsam\AppData\Local\Temp\iv1phtlm.nwz\WorkerExtensions.csproj : error NU1605: Detected package downgrade: System.Net.NameResolution from 4.3.0 to 4.0.0. Reference the package directly from the project to select a different version.
C:\Users\brettsam\AppData\Local\Temp\iv1phtlm.nwz\WorkerExtensions.csproj : error NU1605:  Microsoft.Azure.Functions.Worker.Extensions -> Microsoft.Azure.WebJobs.Extensions.Storage 4.0.4 -> Microsoft.Azure.Cosmos.Table 1.0.7 -> Microsoft.Azure.DocumentDB.Core 2.10.0 -> NETStandard.Library 1.6.0 -> System.Net.Sockets 4.1.0 -> runtime.win.System.Net.Sockets 4.3.0 -> System.Net.NameResolution (>= 4.3.0)
C:\Users\brettsam\AppData\Local\Temp\iv1phtlm.nwz\WorkerExtensions.csproj : error NU1605:  Microsoft.Azure.Functions.Worker.Extensions -> Microsoft.Azure.WebJobs.Extensions.Storage 4.0.4 -> Microsoft.Azure.Cosmos.Table 1.0.7 -> Microsoft.Azure.DocumentDB.Core 2.10.0 -> System.Net.NameResolution (>= 4.0.0)
```

--- a/sdk/Sdk/ExtensionsCsprojGenerator.cs
+++ b/sdk/Sdk/ExtensionsCsprojGenerator.cs
@@ -65,8 +65,7 @@ namespace Microsoft.Azure.Functions.Worker.Sdk
         <Version>$(MajorMinorProductVersion).0</Version>
         <AssemblyVersion>$(MajorMinorProductVersion).0.0</AssemblyVersion>
         <FileVersion>$(Version)</FileVersion>
-        <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
-        <NoWarn>$(NoWarn);NU1605</NoWarn>
+        <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>        
     </PropertyGroup>
     <ItemGroup>
         <PackageReference Include=""Microsoft.NET.Sdk.Functions"" Version=""3.0.11"" />

--- a/sdk/Sdk/ExtensionsCsprojGenerator.cs
+++ b/sdk/Sdk/ExtensionsCsprojGenerator.cs
@@ -68,6 +68,7 @@ namespace Microsoft.Azure.Functions.Worker.Sdk
         <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>        
     </PropertyGroup>
     <ItemGroup>
+        <PackageReference Include=""Microsoft.NETCore.Targets"" Version=""3.0.0"" PrivateAssets=""all"" />
         <PackageReference Include=""Microsoft.NET.Sdk.Functions"" Version=""3.0.11"" />
         {extensionReferences}
     </ItemGroup>

--- a/sdk/Sdk/ExtensionsCsprojGenerator.cs
+++ b/sdk/Sdk/ExtensionsCsprojGenerator.cs
@@ -1,7 +1,7 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-﻿using System;
+using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Text;
@@ -66,6 +66,7 @@ namespace Microsoft.Azure.Functions.Worker.Sdk
         <AssemblyVersion>$(MajorMinorProductVersion).0.0</AssemblyVersion>
         <FileVersion>$(Version)</FileVersion>
         <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
+        <NoWarn>$(NoWarn);NU1605</NoWarn>
     </PropertyGroup>
     <ItemGroup>
         <PackageReference Include=""Microsoft.NET.Sdk.Functions"" Version=""3.0.11"" />

--- a/sdk/Sdk/ExtensionsCsprojGenerator.cs
+++ b/sdk/Sdk/ExtensionsCsprojGenerator.cs
@@ -65,7 +65,7 @@ namespace Microsoft.Azure.Functions.Worker.Sdk
         <Version>$(MajorMinorProductVersion).0</Version>
         <AssemblyVersion>$(MajorMinorProductVersion).0.0</AssemblyVersion>
         <FileVersion>$(Version)</FileVersion>
-        <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>        
+        <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
     </PropertyGroup>
     <ItemGroup>
         <PackageReference Include=""Microsoft.NETCore.Targets"" Version=""3.0.0"" PrivateAssets=""all"" />

--- a/test/FunctionMetadataGeneratorTests/ExtensionsCsProjGeneratorTests.cs
+++ b/test/FunctionMetadataGeneratorTests/ExtensionsCsProjGeneratorTests.cs
@@ -40,10 +40,10 @@ namespace Microsoft.Azure.Functions.SdkTests
         <Version>$(MajorMinorProductVersion).0</Version>
         <AssemblyVersion>$(MajorMinorProductVersion).0.0</AssemblyVersion>
         <FileVersion>$(Version)</FileVersion>
-        <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
-        <NoWarn>$(NoWarn);NU1605</NoWarn>
+        <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>        
     </PropertyGroup>
     <ItemGroup>
+        <PackageReference Include=""Microsoft.NETCore.Targets"" Version=""3.0.0"" PrivateAssets=""all"" />
         <PackageReference Include=""Microsoft.NET.Sdk.Functions"" Version=""3.0.11"" />
         <PackageReference Include=""Microsoft.Azure.WebJobs.Extensions.Storage"" Version=""4.0.3"" />
 <PackageReference Include=""Microsoft.Azure.WebJobs.Extensions.Http"" Version=""3.0.0"" />

--- a/test/FunctionMetadataGeneratorTests/ExtensionsCsProjGeneratorTests.cs
+++ b/test/FunctionMetadataGeneratorTests/ExtensionsCsProjGeneratorTests.cs
@@ -40,7 +40,7 @@ namespace Microsoft.Azure.Functions.SdkTests
         <Version>$(MajorMinorProductVersion).0</Version>
         <AssemblyVersion>$(MajorMinorProductVersion).0.0</AssemblyVersion>
         <FileVersion>$(Version)</FileVersion>
-        <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>        
+        <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
     </PropertyGroup>
     <ItemGroup>
         <PackageReference Include=""Microsoft.NETCore.Targets"" Version=""3.0.0"" PrivateAssets=""all"" />

--- a/test/FunctionMetadataGeneratorTests/ExtensionsCsProjGeneratorTests.cs
+++ b/test/FunctionMetadataGeneratorTests/ExtensionsCsProjGeneratorTests.cs
@@ -1,7 +1,7 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-﻿using System.Collections.Generic;
+using System.Collections.Generic;
 using Microsoft.Azure.Functions.Worker.Sdk;
 using Xunit;
 
@@ -41,6 +41,7 @@ namespace Microsoft.Azure.Functions.SdkTests
         <AssemblyVersion>$(MajorMinorProductVersion).0.0</AssemblyVersion>
         <FileVersion>$(Version)</FileVersion>
         <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
+        <NoWarn>$(NoWarn);NU1605</NoWarn>
     </PropertyGroup>
     <ItemGroup>
         <PackageReference Include=""Microsoft.NET.Sdk.Functions"" Version=""3.0.11"" />

--- a/test/SdkE2ETests/PublishTests.cs
+++ b/test/SdkE2ETests/PublishTests.cs
@@ -21,22 +21,21 @@ namespace Microsoft.Azure.Functions.SdkE2ETests
         }
 
         [Fact]
-        public Task Publish()
-        {
-            return RunPublishTest();
-        }
-
-
-        [Fact]
-        public Task Publish_Rid()
-        {
-            return RunPublishTest("-r win-x86");
-        }
-
-        private async Task RunPublishTest(string additionalParams = null)
+        public async Task Publish()
         {
             string outputDir = await TestUtility.InitializeTestAsync(_testOutputHelper, nameof(Publish));
+            await RunPublishTest(outputDir);
+        }
 
+        [Fact]
+        public async Task Publish_Rid()
+        {
+            string outputDir = await TestUtility.InitializeTestAsync(_testOutputHelper, nameof(Publish_Rid));
+            await RunPublishTest(outputDir, "-r win-x86");
+        }
+
+        private async Task RunPublishTest(string outputDir, string additionalParams = null)
+        {
             // Name of the csproj
             string projectNameToTest = "FunctionApp";
             string projectFileDirectory = Path.Combine(TestUtility.SamplesRoot, projectNameToTest);
@@ -88,8 +87,6 @@ namespace Microsoft.Azure.Functions.SdkE2ETests
             // Verify functions.metadata
             TestUtility.ValidateFunctionsMetadata(functionsMetadataPath, "functions.metadata");
         }
-
-
 
         private class Extension
         {

--- a/test/SdkE2ETests/PublishTests.cs
+++ b/test/SdkE2ETests/PublishTests.cs
@@ -21,7 +21,19 @@ namespace Microsoft.Azure.Functions.SdkE2ETests
         }
 
         [Fact]
-        public async Task Publish()
+        public Task Publish()
+        {
+            return RunPublishTest();
+        }
+
+
+        [Fact]
+        public Task Publish_Rid()
+        {
+            return RunPublishTest("-r win-x86");
+        }
+
+        private async Task RunPublishTest(string additionalParams = null)
         {
             string outputDir = await TestUtility.InitializeTestAsync(_testOutputHelper, nameof(Publish));
 
@@ -38,7 +50,7 @@ namespace Microsoft.Azure.Functions.SdkE2ETests
 
             // Publish
             _testOutputHelper.WriteLine($"[{DateTime.UtcNow:O}] Publishing...");
-            dotnetArgs = $"publish {projectNameToTest}.csproj --configuration {TestUtility.Configuration} -o {outputDir}";
+            dotnetArgs = $"publish {projectNameToTest}.csproj --configuration {TestUtility.Configuration} -o {outputDir} {additionalParams}";
             exitCode = await new ProcessWrapper().RunProcess(TestUtility.DotNetExecutable, dotnetArgs, projectFileDirectory, testOutputHelper: _testOutputHelper);
             Assert.True(exitCode.HasValue && exitCode.Value == 0);
             _testOutputHelper.WriteLine($"[{DateTime.UtcNow:O}] Done.");
@@ -76,6 +88,8 @@ namespace Microsoft.Azure.Functions.SdkE2ETests
             // Verify functions.metadata
             TestUtility.ValidateFunctionsMetadata(functionsMetadataPath, "functions.metadata");
         }
+
+
 
         private class Extension
         {

